metrics/metrics-tests/src/main/java/org/apache/james/metrics/tests/RecordingMetric.java

```diff
@@ -20,36 +20,42 @@
 package org.apache.james.metrics.tests;
 
 import java.util.concurrent.atomic.AtomicInteger;
-import java.util.function.Consumer;
 
 import org.apache.james.metrics.api.Metric;
 
 public class RecordingMetric implements Metric {
     private final AtomicInteger value;
-    private final Consumer<Integer> publishCallback;
 
-    public RecordingMetric(Consumer<Integer> publishCallback) {
-        this.value = new AtomicInteger();
-        this.publishCallback = publishCallback;
+    RecordingMetric() {
+        this(new AtomicInteger());
+    }
+
+    RecordingMetric(AtomicInteger value) {
+        this.value = value;
     }
 
     @Override
     public void increment() {
-        publishCallback.accept(value.incrementAndGet());
+        value.incrementAndGet();
```

Comment:
you don't need anymore `AndGet` in these 4 methods

Reply:
sorry there is no `.increment()` and `decrement()` or something you want from AtomicInteger API